src/SlippiGame.ts

```diff
@@ -72,185 +48,128 @@ export class SlippiGame {
       throw new Error("Cannot create SlippiGame with input of that type");
     }
 
-    this.frameReadPos = null;
-    this.latestFrameIndex = null;
+    // Set up stats calculation
+    this.statsComputer.registerAll([
+      this.actionsComputer,
+      this.comboComputer,
+      this.conversionComputer,
+      this.inputComputer,
+      this.stockComputer,
+    ]);
+    this.parser = new SlpParser(this.statsComputer);
   }
 
-  /**
-   * Gets the game settings, these are the settings that describe the starting state of
-   * the game such as characters, stage, etc.
-   */
-  getSettings(): GameStartType {
-    if (this.settings) {
-      // If header is already generated, return it
-      return this.settings;
+  private _process(settingsOnly = false): void {
+    if (this.parser.getGameEnd() !== null) {
+      return;
     }
-
     const slpfile = openSlpFile(this.input);
-
-    // Prepare default settings
-    let settings: GameStartType = null;
-
     // Generate settings from iterating through file
-    iterateEvents(slpfile, (command, payload) => {
+    this.readPosition = iterateEvents(slpfile, (command, payload) => {
       if (!payload) {
         // If payload is falsy, keep iterating. The parser probably just doesn't know
         // about this command yet
         return false;
       }
 
       switch (command) {
-      case Command.GAME_START:
-        payload = payload as GameStartType;
-        if (!payload.stageId) {
-          return true; // Why do I have to do this? Still not sold on Flow
-        }
+        case Command.GAME_START:
```

Comment:
I personally prefer the previous style of indenting. It kind of bothers me that with this form, switch statements double-indent logic whereas an `if/else` statement would have only single indented the logic. Do you have a preference? If so why?

Reply:
Nope. I used the built-in auto formatter. I'm sure there's some style guide or some standard that the formatter is using so I just used it to clean up the code. No particular reason but imo using the auto-formatter is easy so I use it.

Reply:
Yeah, agreed. Do you know if the auto-formatter uses the eslint rules or something?

Reply:
No idea. Probably some vscode setting or something though. Haven't looked into it.